### PR TITLE
NAS-114383 / 13.0 / NAS-114383: enclosure plugin rewritten on 13 needs webUI changes

### PR DESCRIPTION
**Pull request by bvasilenko**
Summary: adapted response of `EnclosureData` to existing UI logic.

Testing:

To see **System -> View Enclosures**:
1. edit src/app/services/navigation/navigation.service.ts
   image

2. edit src/app/pages/system/viewenclosure/view-enclosure.component.ts:
   image

3. remove condition at src\\app\\pages\\system\\viewenclosure\\view-enclosure.component.html:2
   image

   
To see contents of **View Enclosures**:
1. Download new JSON response for  `EnclosureData` query from attachments of the ticket https://jira.ixsystems.com/browse/NAS-114383
2. Extract responses  `PoolData`, `SensorData` and `DisksData` queres from the machine mentioned in the ticket https://jira.ixsystems.com/browse/NAS-114383
3. Paste JSON to src/app/pages/system/viewenclosure/view-enclosure.component.ts  
   image

### CODE REVIEW

**Comment by undsoft** [comment]
@bvasilenko Isn't there a real machine that can be used for testing?

**Comment by bvasilenko** [comment]
There is a machine mentioned in the ticket.
> from the machine mentioned in the ticket https://jira.ixsystems.com/browse/NAS-114383

File: src/app/pages/system/viewenclosure/view-enclosure.component.ts [hunk 1]
```diff
@@ -270,4 +272,31 @@ export class ViewEnclosureComponent implements AfterContentInit, OnChanges, OnDe
       this.currentView = disks;
     }
   }
+
+  private _parseEnclosureData(enclosure) {
```
Comment (RehanY147) [inline src/app/pages/system/viewenclosure/view-enclosure.component.ts:276]: Should have type for the parameter and the name shouldn't start with `_`

File: src/app/pages/system/viewenclosure/view-enclosure.component.ts [hunk 2]
```diff
@@ -270,4 +272,31 @@ export class ViewEnclosureComponent implements AfterContentInit, OnChanges, OnDe
...
+    const parsedEnclosure = [];
+
+    enclosure.forEach((e, idx) => {
```
Comment (RehanY147) [inline src/app/pages/system/viewenclosure/view-enclosure.component.ts:279]: One letter variable name should be avoided.

File: src/app/pages/system/viewenclosure/view-enclosure.component.ts [hunk 3]
```diff
@@ -270,4 +272,31 @@ export class ViewEnclosureComponent implements AfterContentInit, OnChanges, OnDe
       this.currentView = disks;
     }
   }
+
+  private _parseEnclosureData(enclosure) {
+    const parsedEnclosure = [];
```
Comment (RehanY147) [inline src/app/pages/system/viewenclosure/view-enclosure.component.ts:277]: Should have a type

**Review by RehanY147 — CHANGES_REQUESTED** [review]
The changes work but I can't see the difference between the normal system or post changes. What is the problem behavior that we're trying to fix? And how? @bvasilenko Please provider screenshots. Could I simply test this on the machine mentioned on the ticket? Or do I need to do all of the steps described above to test it?

**Comment by bvasilenko** [comment]
Here's the screenshots depicting how WebUI looks during testing with mockup responses (which I provided instructions for above at the description):

Before my fix (`parseEnclosureData`)
image

After my fix:
image

Testing on real machine: We actually don't have a machine to test with. The machine we currently have is v12 does not produce the response in the expected format. Instead, it produces the response in the old format. We need a machine with middleware v13 for testing on the hardware. I've also left a comment here: https://jira.ixsystems.com/browse/NAS-114383?focusedCommentId=148512&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-148512

**Comment by bvasilenko** [comment]
@RehanY147 
Update: here's the machine for testing [tn24.lab.ixsystems.net](http://tn24.lab.ixsystems.net/) user: `root` pass: `abcd1234`

**Review by RehanY147 — APPROVED** [review]
Looks good. The messy view enclosure is fixed now.

**Pull request merged**